sky/serve/service_spec.py

```diff
@@ -45,8 +53,45 @@ def __init__(
         self._max_replicas = max_replicas
         self._qps_upper_threshold = qps_upper_threshold
         self._qps_lower_threshold = qps_lower_threshold
+        self._target_qps_per_replica = target_qps_per_replica
         self._post_data = post_data
         self._auto_restart = auto_restart
+        spot_args = [
+            spot_placer, spot_mixer, spot_zones, target_qps_per_replica,
+            num_extra
+        ]
+        on_demand_args = [target_qps_per_replica, on_demand_type]
+        spot_args_num = sum([spot_arg is not None for spot_arg in spot_args])
+        on_demand_args_num = sum(
+            [on_demand_arg is not None for on_demand_arg in on_demand_args])
+        if not spot_placer:
+            if (on_demand_args_num != 0 and
+                    on_demand_args_num != len(on_demand_args)):
+                with ux_utils.print_exception_no_traceback():
+                    raise ValueError('target_qps_per_replica and'
+                                     'on_demand_type must be all '
+                                     'specified or all not specified'
+                                     'in the service YAML.')
+        if spot_placer:
+            if spot_args_num != 0 and spot_args_num != len(
+                    spot_args) and on_demand_args_num == 0:
+                with ux_utils.print_exception_no_traceback():
+                    raise ValueError(
+                        'spot_placer, spot_mixer, '
+                        'target_qps_per_replica, num_extra '
+                        'and spot_zones must be all '
+                        'specified or all not specified in the service YAML.')
+        # TODO(tian): Check vanilla autoscaler argument not exist if spot
+        # TODO(tian): Warning user that the use_spot in resources is ignored
```

Comment:
modified. let's raise an error if the user specifies 'use_spot: False' and spot_placer is enabled.